dora/core/common/src/main/java/alluxio/wire/WorkerIdentity.java

```diff
@@ -118,6 +120,33 @@ public String toString() {
         Parsers.getParserOfVersion(mVersion).getVersionSpecificRepresentation(this));
   }
 
+  /**
+   * Construct from a workerid representation from registration.
+   *
+   * @param workerIdentityStr
+   * @return WorkerIdentity object
+   * @throws InvalidArgumentException
+   */
+  public static WorkerIdentity fromString(String workerIdentityStr)
+      throws InvalidArgumentException {
+    String prefix = "worker-";
+    String errStr = "Unrecognized worker identity string.";
+    if (!workerIdentityStr.startsWith(prefix)) {
+      throw new InvalidArgumentException(errStr);
+    }
```

Comment:
I don't think missing the `worker-` prefix warrants an error, i.e. both `worker-1` and `1` should be considered valid worker IDs.

Reply:
i actually wanted to make sure tostring and fromstring are strictly paired, i've also added the warning comment should someone wants to change it.